policy/reviewer/reviewer.go

```diff
@@ -124,31 +127,94 @@ func FindRandomRequesters(ctx context.Context, prctx pull.Context, result common
 			shoveIntoMap(allUsers, orgMembers)
 		}
 
-		allCollaboratorPermissions, err := prctx.RepositoryCollaborators()
+		allCollaboratorPermissions := make(map[string]string)
+		allCollaboratorAndPermissions, err := prctx.RepositoryCollaborators()
 		if err != nil {
 			return nil, errors.Wrap(err, "Unable to list repository collaborators")
 		}
 
-		if child.ReviewRequestRule.Admins {
-			var repoAdmins []string
-			for user, perm := range allCollaboratorPermissions {
-				if perm == common.GithubAdminPermission {
-					repoAdmins = append(repoAdmins, user)
-				}
-			}
-			shoveIntoMap(allUsers, repoAdmins)
-		}
-
 		if child.ReviewRequestRule.WriteCollaborators {
 			var repoCollaborators []string
-			for user, perm := range allCollaboratorPermissions {
+			for user, perm := range allCollaboratorAndPermissions {
 				if perm == common.GithubWritePermission {
 					repoCollaborators = append(repoCollaborators, user)
 				}
 			}
 			shoveIntoMap(allUsers, repoCollaborators)
 		}
 
+		// When admins are selected for review, only collect the desired set of admins instead of
+		// everyone, which includes org admins
+		if !child.ReviewRequestRule.Admins {
+			// When not looking for admins, we want to check with everyone
+			allCollaboratorPermissions = allCollaboratorAndPermissions
+		} else {
+			// Determine what the scope of requested admins should be
```

Comment:
This feels significant enough to merit a separate helper function, maybe `selectAdmins` to match `selectTeamMembers` and `selectOrgMembers`.